src/main/java/jp/ac/utokyo/rcast/karkinos/readssummary/GeneExons.java

```diff
@@ -75,93 +86,72 @@ public void setTumortotal(long tumortotal) {
 	}
 
 	Map<String,Integer> counter = new HashMap<String,Integer>();
-	private void loadmap(String refflat) {
-		// TODO Auto-generated method stub
-		try {
-			CSVReader brcvs = new CSVReader(new FileReader(refflat), '\t');
-
-			String[] data = null;
-			while ((data = brcvs.readNext()) != null) {
-
-				try {
-					
-					String geneSymbol = data[0];
-					String symbolWithoutNum = symbolWithoutNum(geneSymbol);
-					//System.out.println(geneSymbol+"\t"+symbolWithoutNum);
-					if(counter.containsKey(symbolWithoutNum)){
-						int countn = counter.get(symbolWithoutNum);
-						countn++;
-						counter.put(symbolWithoutNum, countn);
-					}else{
-						counter.put(symbolWithoutNum, 1);
-					}
-					
-					
-					String refseqid = data[1];
-					String chrom = data[2];
-					int txstart = Integer.parseInt(data[4]);
-					int txend = Integer.parseInt(data[5]);
-					TreeMap<Integer, Interval> tm0 = genemap.get(chrom);
-					if (tm0 == null) {
-						tm0 = new TreeMap<Integer, Interval>();
-						genemap.put(chrom, tm0);
-					}
-					Interval iv0 = new Interval(chrom, txstart, txend, refseqid,geneSymbol);
-					tm0.put(txstart, iv0);
-
-					int cdsstart = Integer.parseInt(data[6]);
-					int cdsend = Integer.parseInt(data[7]);
-					String exonstarts = data[9];
-					String exonends = data[10];
-					List<Integer> es = toIntList(exonstarts);
-					List<Integer> ee = toIntList(exonends);
-					// //
+
+	private void loadmap(final String refflat) {
+		try (final CSVReader brcvs = new CSVReader(new FileReader(refflat), '\t')) {
+			try {
+				String[] data = null;
+				while ((data = brcvs.readNext()) != null) {
+					final String geneSymbol = data[0];
+					final String symbolWithoutNum = symbolWithoutNum(geneSymbol);
+					this.counter.put(symbolWithoutNum, 1 + this.counter.getOrDefault(symbolWithoutNum, 0));
+
+					// Convert refGene.txt format (0-based half-opened range)
+					// into Interval format (1-based closed open range).
+					final String refseqid = data[1];
+					final String chrom = data[2];
+					final int txStart = Integer.parseInt(data[4]);
+					final int txEnd = Integer.parseInt(data[5]);
+					this.genemap
+					    .computeIfAbsent(chrom, k -> new TreeMap<Integer, Interval>())
```

Comment:
Please use the diamond operator.

```suggestion
					    .computeIfAbsent(chrom, k -> new TreeMap<>())
```